packages/blockchain-wallet-v4-frontend/src/components/Flyout/FrequencyScreen.tsx

```diff
@@ -2,25 +2,72 @@ import React, { useCallback } from 'react'
 import { FormattedMessage } from 'react-intl'
 
 import { Text } from 'blockchain-info-components'
+import { SBPaymentTypes } from 'blockchain-wallet-v4/src/network/api/simpleBuy/types'
 
-import { RecurringBuyPeriods } from '../../data/components/recurringBuy/types'
+import {
+  RecurringBuyNextPayment,
+  RecurringBuyPeriods
+} from '../../data/components/recurringBuy/types'
 import { OptionRightActionRow } from '../Rows'
 import Container from './Container'
 import Content from './Content'
 import Header, { Props as HeaderProps } from './Header'
 import { getPeriodSubTitleText, getPeriodTitleText } from './model'
 
-const FrequencyScreen = ({ children, headerAction, headerMode, setPeriod }: Props) => {
-  // ONE_TIME is not a recurring buy option so take it out before displaying
-  const periods = Object.values(RecurringBuyPeriods)
+const RowDisplay = ({ method, paymentInfo, period, setPeriod }: RowDisplayProps) => {
+  // ONE_TIME is a special case that we need to insert manually
```

Comment:
can you expand on this comment for future us? when is ONE_TIME used and when is it not?

Reply:
Gostei